### DISCUSSION TITLE
[BUG] intermittent Microsoft pop up after Login

### DISCUSSION THREAD

**Discussion by Abhilash-Swain** [pnp/powershell#2393] [General]
### Notice
Many bugs reported are actually related to the PnP Framework which is used behind the scenes. Consider carefully where to report an issue:

1. **Are you using ```Invoke-PnPSiteTemplate``` or ```Get-PnPSiteTemplate```**? The issue is most likely related to the Provisioning Engine. The Provisioning engine is _not_ located in the PowerShell repo. Please report the issue here: https://github.com/pnp/pnpframework/issues.
1. **Is the issue related to the cmdlet itself, its parameters, the syntax, or do you suspect it is the code of the cmdlet that is causing the issue?** Then please continue reporting the issue in this repo.
1. **If you think that the functionality might be related to the underlying libraries that the cmdlet is calling** (We realize that might be difficult to determine), please first double check the code of the cmdlet, which can be found here: https://github.com/pnp/powershell/tree/master/src/Commands. If related to the cmdlet, continue reporting the issue here, otherwise report the issue at https://github.com/pnp/pnpframework/issues

### Reporting an Issue or Missing Feature
Getting intermittent Microsoft Login pop up in PowerShell while trying to download a file  . (We complete the login activity before trying to download. )

FYI:- We are using PnP in an unatteneded automation. The 1st step is login to PnP and then download the file with a download script.
(Both scripts listed below for your review) 
**Login Script : -**
Invoke-Expression "Cmd.exe /C RunDll32.exe InetCpl.cpl,ClearMyTracksByProcess 2"  (##To clear computer cache and force for login)
Connect-PnPOnline -Url "My Sharepoint URL" -UseWebLogin

**File Download: -**

Connect-PnPOnline -Url "My SharePoint URL2" -UseWebLogin (##Reconnecting to a different Site/Subsite in same server)
Get-PnPFile -AsFile -Url  -Filename $sourceitem.Name -Path "C:\Apps\AB"

### Expected behavior 
An Intermittent Microsoft login pop up should not appear while downloading.
We are expecting same behavior in new PnP.powershell Module 

### Actual behavior 
![MicrosoftTeams-image](https://user-images.githubusercontent.com/101560357/192352917-e7c40389-8cc6-4b08-a279-6dd4df7d8be1.png)

Getting this popup while trying to run download script shared above.

### Steps to reproduce behavior
The scripts running in unattended mode and the issue happens intermittently . Hence it is difficult to reproduce.
However if the login is not success (Step -1 Sp login script run), the download script prompt  Microsoft login window.

We suspect if by any chance the SharePoint login session is getting deactivated within few minutes. This will also result the login pop up screen again while running the Download script.

### What is the version of the Cmdlet module you are running?
(you can retrieve this by executing ```Get-Module -Name "PnP.PowerShell" -ListAvailable```)

The current powershell module is "SharePointPnPPowerShellOnline " V "3.26.2010.0"
Migrating to PnP.Powershell V 1.10

### Which operating system/environment are you running PnP PowerShell on?
- [ yes] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by veronicageek** [comment]
@216-Abhi - Can you please feedback when you try with the module `1.11` (latest) and possibly `1.10`. The authentication has changed between `1.10` and `1.11` so that will help troubleshooting if one work but not the other.

If you're not familiar with the new PnP PowerShell module, this will get you started: https://pnp.github.io/powershell/articles/authentication.html

With regards to `SharePointPnPPowerShellOnline 3.26.2010.0`, we no longer support this version/repo and therefore we don't update following any changes MSFT would make.

**Comment by Abhilash-Swain** [comment]
Hi @veronicageek ,
Thanks for reverting.

We are trying to authenticate using "-UseWebLogin" feature  as it handles our single sign on correctly.
Is there any better approach to handle single sign on in PnP.PowerShell?

Also, could you please confirm , Once the new version available for PnP, will it auto upgrade my installed version? Ex:- if I have V1.10 installed and after V1.11 available, will it auto upgrade my installed version to V1.11 ? If yes, then how to stop this ?

**Comment by gautamdsheth** [comment]
For unattended logins, would suggest using the client id + certificate issues, check this link: 

https://pnp.github.io/powershell/articles/connecting.html#connect-using-a-clientid-and-pfx-certificate-stored-on-your-local-machine

Weblogin is a legacy way of authentication, would suggest usage of certificate approach.
Also, this looks like an issue with your script not working and not really an issue with the PnP PowerShell module, so converting this to a discussion and closing it here.
